cmd/uplink/cmd/setup.go

```diff
@@ -251,3 +260,37 @@ func ApplyDefaultHostAndPortToAddr(address, defaultAddress string) (string, erro
 	// address is host:
 	return net.JoinHostPort(addressParts[0], defaultPort), nil
 }
+
+// SaveEncryptionKey saves the key in a new file which will be stored in
+// filepath.
+// It returns an error if the directory doesn't exist, the file already exists
+// or there is an I/O error.
+func SaveEncryptionKey(key []byte, filepath string) (err error) {
+	f, err := os.OpenFile(filepath, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0600)
```

Comment:
`ioutil.WriteFile`

Reply:
I don't think that we want to truncate the file if already exists; currently the function return an error if the file already exists.